### PR TITLE
Better error message for Expression.__nonzero__

### LINKED ISSUE #2764: Add more examples to the Expression.__nonzero__ error
Add example of using `and` and `not`

### PR DESCRIPTION
**Pull request by tpoterba**
fixes #2764

### CODE REVIEW

File: python/hail/expr/expression.py [hunk 1]
```diff
@@ -326,7 +326,10 @@ def _init(self):
 
     def __nonzero__(self):
         raise NotImplementedError(
-            "The truth value of an expression is undefined\n  Hint: instead of if/else, use 'f.cond'")
+            "The truth value of an expression is undefined\n"
+            "    Hint: instead of 'if x', use 'functions.cond(x, ...)'\n"
+            "    Hint: instead of 'x and y' or 'x or y', use 'x & y' or 'x | y'"
```
Comment (jbloom22) [inline python/hail/expr/expression.py]: you need a newline here

File: python/hail/expr/expression.py [hunk 2]
```diff
@@ -326,7 +326,10 @@ def _init(self):
...
+            "    Hint: instead of 'not x', use ~x")
```
Comment (jbloom22) [inline python/hail/expr/expression.py]: and put ~x in single quotes too

**Comment by tpoterba** [comment]
done

File: python/hail/expr/expression.py [hunk 3]
```diff
@@ -326,7 +326,10 @@ def _init(self):
 
     def __nonzero__(self):
         raise NotImplementedError(
-            "The truth value of an expression is undefined\n  Hint: instead of if/else, use 'f.cond'")
+            "The truth value of an expression is undefined\n"
```
Comment (jbloom22) [inline python/hail/expr/expression.py]: Add period? What's your convention.

Comment (tpoterba) [reply python/hail/expr/expression.py]: generally no period for the first line, that's what Python does.

**Pull request merged**